packages/wdio-browserstack-service/src/service.js

```diff
@@ -57,32 +67,51 @@ export default class BrowserstackService {
             `${test.parent} - ${test.title}`
         )
 
-        if (!results.passed) {
-            this.failures++
-            this.failReason = (test.error && test.error.message ? test.error.message : 'Unknown Error')
+        if (!passed) {
+            this.failReasons.push((error && error.message) || 'Unknown Error')
         }
     }
 
-    after() {
-        return this._update(this.sessionId, this._getBody())
+    after(result) {
+        // For Cucumber: Checks scenarios that ran (i.e. not skipped) on the session
+        // Only 1 Scenario ran and option enabled => Redefine session name to Scenario's name
+        if(this.preferScenarioName && this.scenariosThatRan.length === 1){
+            this.fullTitle = this.scenariosThatRan.pop()
+        }
+
+        const hasReasons = Boolean(this.failReasons.filter(Boolean).length)
+
+        return this._update(this.sessionId, {
+            status: result === 0 ? 'passed' : 'failed',
+            name: this.fullTitle,
+            reason: hasReasons ? this.failReasons.join('\n') : undefined
+        })
     }
 
     /**
      * For CucumberJS
      */
 
-    afterScenario(uri, feature, pickle, result) {
-        if (result.status === 'failed') {
-            ++this.failures
+    afterScenario(uri, feature, pickle, { exception, status }) {
+        if (status !== 'skipped') {
+            this.scenariosThatRan.push(pickle.name)
+        }
+
+        if (this.failureStatuses.includes(status)) {
+            exception = exception || (status === 'pending'
+                ? `Some steps/hooks are pending for scenario "${pickle.name}"`
+                : 'Unknown Error')
+
+            this.failReasons.push(exception)
         }
     }
 
     async onReload(oldSessionId, newSessionId) {
         this.sessionId = newSessionId
-        await this._update(oldSessionId, this._getBody())
-        this.failures = 0
+        await this._update(oldSessionId, { name: this.fullTitle })
```

Comment:
how is the status updated in this case?

Reply:
@erwinheitzman  @christian-bromann
I took a look at the Sauce service:
https://github.com/webdriverio/webdriverio/blob/e30d665b06f9a1b16dcf721471fd2488339d67ea/packages/wdio-sauce-service/src/service.js#L11

apparently the whole raison d'être for the failure counter (which this PR removes) is so that  `onReload` hook can report on the ending session

We don't have any session results exposed on `onReload`, we would ideally use what is later given in `after` hook. Would it make sense to expose some `result`s on the `onReload` hook along with `oldSessionId` and `newSessionId`? Something like 
```js
    onReload: function(oldSessionId, result, newSessionId) {
    ...
    },
```

Illustrating the problem:
```js
it("Test that runs first", () => throw new Error('I fail!')) // sessionId 1 will fail
it("Test that runs second", () => {
    console.log("I am running on session 1")
    browser.reloadSession() // We should update session 1 with status "failed"
    // but there's no session 1 results exposed in `onReload`
    // every service has to count failures so far (with varying degrees of success)
    return true // session 2 passes and `after` will update the status accordingly
})
```